### PR TITLE
qa/tests: clients' upgrades for older releases must use "-t py2"

### PR DESCRIPTION
**Pull request by yuriw**
Fixes: https://tracker.ceph.com/issues/47120
Signed-off-by: Yuri Weinstein <yweinste@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by kshtsk — APPROVED** [review]
Looks fine.

File: qa/crontab/teuthology-cronjobs [hunk 1]
```diff
@@ -167,9 +167,9 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
...
-50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade-clients/client-upgrade-nautilus-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch nautilus
+47 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-luminous-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch luminous -t py2
+50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-mimic-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch mimic -t py2
+50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade-clients/client-upgrade-nautilus-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch nautilus -t py2
```
Comment (neha-ojha) [inline qa/crontab/teuthology-cronjobs:172]: why do we need it with nautilus?

Comment (yuriw) [reply qa/crontab/teuthology-cronjobs:172]: @neha-ojha 
see https://pulpito.ceph.com/?suite=upgrade-clients%3Aclient-upgrade-nautilus-octopus

Comment (neha-ojha) [reply qa/crontab/teuthology-cronjobs:172]: @yuriw those are already passing since nautilus works fine with -t master, so we don't need -t py2 here

**Review by tchaikov — APPROVED** [review]

File: qa/crontab/teuthology-cronjobs [hunk 2]
```diff
@@ -119,9 +119,9 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
 ## --filter "ubuntu_16.04,ubuntu_18.04,centos_7.6,rhel_7.6" - test ONLY supported distro AFTER mimic
 ## The suites below run withoit filters
 
-47 01 * * 3  CEPH_BRANCH=nautilus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-jewel-nautilus -k distro -e $CEPH_QA_EMAIL --suite-branch jewel
-50 01 * * 3  CEPH_BRANCH=nautilus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-luminous-nautilus -k distro -e $CEPH_QA_EMAIL --suite-branch luminous
-50 01 * * 3  CEPH_BRANCH=nautilus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-mimic -k distro -e $CEPH_QA_EMAIL --suite-branch mimic
+47 01 * * 3  CEPH_BRANCH=nautilus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-jewel-nautilus -k distro -e $CEPH_QA_EMAIL --suite-branch jewel -t py2
```
Comment (tchaikov) [inline qa/crontab/teuthology-cronjobs:122]: why are we still running upgrade tests with the combination of nautilus and jewel ? i think we are not touching them anymore. so if the tests passed before, they should still pass now.

**Review by tchaikov — CHANGES_REQUESTED** [review]

File: qa/crontab/teuthology-cronjobs [hunk 3]
```diff
@@ -167,8 +167,8 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
 ## !!!! three suites below MUST use --suite-branch luminous, mimic, nautilus (see https://tracker.ceph.com/issues/24021)
 ## The suites below run withoit filters
 
-47 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-luminous-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch luminous
-50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-mimic-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch mimic
+47 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-luminous-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch luminous -t py2
+50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 7 -m $MACHINE_NAME -s upgrade/client-upgrade-mimic-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch mimic -t py2
```
[the remark this answers is not in the document]
Comment (yuriw) [reply qa/crontab/teuthology-cronjobs:171]: this has been like that forever

File: qa/crontab/teuthology-cronjobs (hunk 2 above)
Comment (yuriw) [reply qa/crontab/teuthology-cronjobs:122]: yes

**Comment by yuriw** [comment]
@tchaikov pls approve, we want merge this as test failing, we can undo later if more elegant solution is available.  I prefer using explicit `-t` in schedules so we don;t rely on defaults, but open for any.

File: qa/crontab/teuthology-cronjobs (hunk 2 above)
Comment (tchaikov) [reply qa/crontab/teuthology-cronjobs:122]: i don't understand what "yes" stands for.

File: qa/crontab/teuthology-cronjobs (hunk 3 above)
[the remark this answers is not in the document]
Comment (tchaikov) [reply qa/crontab/teuthology-cronjobs:171]: this does not resolve the issue.

File: qa/crontab/teuthology-cronjobs (hunk 2 above)
Comment (tchaikov) [reply qa/crontab/teuthology-cronjobs:122]: sorry, i was wrong. nautilus is not EOL yet.

**Pull request merged**